.github/workflows/docker.yml

```diff
@@ -64,16 +64,24 @@ jobs:
           echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
           cat $GITHUB_OUTPUT
 
-      - name: Prepare Go Build Cache for Docker
-        uses: actions/cache@v3
+      - uses: actions/setup-go@v5
         with:
-          path: go-build-cache
-          key: ${{ runner.os }}-go-build-cache-${{ hashFiles('**/go.sum') }}
+          cache-dependency-path: "**/*.sum"
+          go-version-file: 'temporal/go.mod'
 
-      - name: Inject go-build-cache into docker
-        uses: reproducible-containers/buildkit-cache-dance@v2.1.3
-        with:
-          cache-source: go-build-cache
+      - name: Compile binaries
+        run: make bins
+
+      - name: Get native architecture
+        run: |
+          case ${RUNNER_ARCH} in
+            X*)
```

Comment:
what are the possible values we expect here?

Reply:
According to [GitHub's docs](https://docs.github.com/en/actions/learn-github-actions/variables):
> `RUNNER_ARCH`	The architecture of the runner executing the job. Possible values are X86, X64, ARM, or ARM64